lib/mas/firm_indexer.rb

```diff
@@ -0,0 +1,32 @@
+module FirmIndexer
+  class << self
+    def index_firm(firm)
+      if !firm.destroyed? && firm.publishable?
+        store_firm(firm)
+      else
+        delete_firm(firm)
+      end
+    end
+
+    def handle_firm_changed(firm)
+      index_firm(firm)
+    end
+
+    def handle_aggregate_changed(aggregate)
+      index_firm(aggregate.firm) if firm_exists?(aggregate.firm)
+    end
+
+    def store_firm(firm)
```

Comment:
I see what you mean. They remained public to this point because I wanted to be able to test `firm_exists?` so I could simplify the test for `handle_aggregate_changed?` and mock store/delete_firm. I'll have a rethink at this point and see.